server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1910,12 +1911,11 @@ public void setDocIdFuzzySetFalsePositiveProbability(double docIdFuzzySetFalsePo
 
     public RemoteStorePathStrategy getRemoteStorePathStrategy() {
         Map<String, String> remoteCustomData = indexMetadata.getCustomData(IndexMetadata.REMOTE_STORE_CUSTOM_KEY);
-        if (remoteCustomData != null
-            && remoteCustomData.containsKey(PathType.NAME)
-            && remoteCustomData.containsKey(PathHashAlgorithm.NAME)) {
+        if (remoteCustomData != null && remoteCustomData.containsKey(PathType.NAME)) {
             PathType pathType = PathType.parseString(remoteCustomData.get(PathType.NAME));
-            PathHashAlgorithm pathHashAlgorithm = PathHashAlgorithm.parseString(remoteCustomData.get(PathHashAlgorithm.NAME));
-            return new RemoteStorePathStrategy(pathType, pathHashAlgorithm);
+            String hashAlgoStr = remoteCustomData.get(PathHashAlgorithm.NAME);
+            PathHashAlgorithm hashAlgorithm = Objects.nonNull(hashAlgoStr) ? PathHashAlgorithm.parseString(hashAlgoStr) : null;
+            return new RemoteStorePathStrategy(pathType, hashAlgorithm);
```

Comment:
if pathType is set and hashAlgorithm is not set and passed as null, will it use FIXED path strategy?

Reply:
We had an assertion earlier, but **now i have replaced with an exception**. This should never happen that pathType is hash based and there is no path hash algorithm. Such cases are bugs and should be fixed.

Reply:
should we memoize`RemoteStorePathStrategy` in index settings ?

Reply:
Done

Reply:
If we are using the hashed prefix, then index metadata would be updated before the first shard starts on remote node.  So i don't think we would need changes . LMK if i am missing something

Reply:
> If we are using the hashed prefix, then index metadata would be updated before the first shard starts on remote node.
Not required for this case. If we are updating index metadata at the end, then we do need it.